packages/amplify_flutter/lib/amplify.dart

```diff
@@ -158,15 +170,20 @@ class AmplifyClass extends PlatformInterface {
           underlyingException: e.toString());
     }
     try {
+      _config = _createConfigObject(configuration);
```

Comment:
This will throw an exception if a value that we expect to be non-null is null, correct?

Is there a spec or type definition for the config that you were able to reference to determine which values are optional/required?

How do we handle breaking changes? If the CLI changes a required property to be optional, would the CLI treat that as a breaking change and create a new major version? Would we then have to tell customers to not upgrade to a new version of the CLI until they are using version x.y.z of amplify-flutter?

I guess the iOS/Android/JS libraries are already dealing with this to an an extent, but nullability works differently in Java/Swift/TS. If something like `authenticationFlowType` hypothetically became optional in a future CLI version, an exception would be thrown trying to read the config since it is cast to a `String`, even though that value is never accessed in amplify-flutter or the authenticator. Is that true for  iOS/Android?

I know some of this was already discussed, so sorry if I am asking questions that have already been answered.

Reply:
It would be good to provide meaningful exceptions in the case that there is an exception reading the config. Inevitably some customers are going to have issues with their config missing properties that we are expecting to be non-null (either because they are using an outdated version of the CLI, they are using a new version of the CLI with breaking changes, or because we simply marked something as non-nullable that should not have been).

The call to `_createConfigObject` could be wrapped in its own try catch, with a meaningful message.

We might event want to consider calling completeError and not throwing an exception if the config cannot be parsed successfully since it is only required for using the authenticator at the moment. In a scenario where a customer who is not using the authenticator has a config that iOS/Android can parse, but amplify-flutter cannot, it would be a poor experience to throw an exception here. If we have a spec/type definition for the config that we feel confident in, it is probably fine to throw. I just want to avoid impacting customers who aren't event using the authenticator.

Reply:
As we discussed, we are not going to use completeError, and are going to rely on a) native libraries catching major problems with the config, and b) making all config values nullable so that `_createConfigObject` should not fail.

Reply:
I suppose we could wrap the call to  _createConfigObject in a try/catch, but at some point I think we would need an empty catch block because we don't want to propogate the exception all the way up?